diff --git a/3rdparty/perfparser.cmake b/3rdparty/perfparser.cmake
index 18b62b3..c45a83a 100644
--- a/3rdparty/perfparser.cmake
+++ b/3rdparty/perfparser.cmake
@@ -41,6 +41,7 @@ PUBLIC
     Qt::Network
     ${LIBDW_LIBRARIES}
     ${LIBELF_LIBRARIES}
+    iberty
 )
 
 if (Zstd_FOUND)
diff --git a/CMakeLists.txt b/CMakeLists.txt
index b6885c2..f924a98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,6 +149,11 @@ include(ECMAddTests)
 include(ECMCoverageOption)
 include(ECMEnableSanitizers)
 
+set(BREEZE_EXTENSIONS all CACHE STRING "The extensions to include in our stylesheets.")
+set(BREEZE_STYLES all CACHE STRING "The styles to include in our stylesheets.")
+
+include(breeze)
+
 file(RELATIVE_PATH LIBEXEC_REL_PATH "${KDE_INSTALL_FULL_BINDIR}" "${KDE_INSTALL_FULL_LIBEXECDIR}")
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/hotspot-config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/hotspot-config.h @ONLY)
diff --git a/cmake/breeze.cmake b/cmake/breeze.cmake
new file mode 100644
index 0000000..8841824
--- /dev/null
+++ b/cmake/breeze.cmake
@@ -0,0 +1,53 @@
+# Setup Qt: this works with both Qt5 and Qt6
+# NOTE: We use cached strings to specify the options for these.
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
+
+find_package(
+  Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION}
+  COMPONENTS Core Gui Widgets Svg
+  REQUIRED)
+# -------------------
+
+# Get Python to compile the stylesheets.
+# Fetch the repository, configure, compile the stylesheets.
+find_package(Python COMPONENTS Interpreter)
+
+include(FetchContent)
+
+set(FETCHCONTENT_QUIET OFF CACHE BOOL "Silence fetch content" FORCE)
+set(PREFER_PIE OFF CACHE BOOL "If to prefer position-independent code, if the compiler supports it.")
+set(BREEZE_EXTENSIONS all CACHE STRING "The extensions to include in our stylesheets.")
+set(BREEZE_STYLES all CACHE STRING "The styles to include in our stylesheets.")
+
+FetchContent_Declare(
+  breeze_stylesheets
+  GIT_REPOSITORY https://github.com/Alexhuszagh/BreezeStyleSheets.git
+  GIT_TAG origin/main
+  GIT_PROGRESS ON
+  GIT_SHALLOW 1
+  USES_TERMINAL_DOWNLOAD TRUE)
+
+FetchContent_MakeAvailable(breeze_stylesheets)
+
+add_library(breeze STATIC "${breeze_stylesheets_SOURCE_DIR}/dist/styles/breeze.qrc")
+
+add_custom_target(
+run_python_breeze ALL
+COMMAND ${Python_EXECUTABLE} configure.py --extensions=${BREEZE_EXTENSIONS}
+        --styles=${BREEZE_STYLES} --resource breeze.qrc
+WORKING_DIRECTORY ${breeze_stylesheets_SOURCE_DIR}
+BYPRODUCTS "${breeze_stylesheets_SOURCE_DIR}/dist/styles/breeze.qrc"
+COMMENT "Generating themes")
+
+add_dependencies(breeze run_python_breeze)
+
+# Prefer position independent code, if the compiler supports it.
+if(PREFER_PIE)
+  include(CheckPIESupported)
+  check_pie_supported(LANGUAGES CXX OUTPUT_VARIABLE pie-supported)
+  if(CMAKE_C_LINK_PIE_SUPPORTED)
+    set_property(TARGET breeze PROPERTY POSITION_INDEPENDENT_CODE ON)
+  endif()
+endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7424533..54a4991 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -118,6 +118,7 @@ target_link_libraries(
     models
     PrefixTickLabels
     KDAB::kddockwidgets
+    breeze
 )
 
 if(KFArchive_FOUND)
diff --git a/src/main.cpp b/src/main.cpp
index 1da2636..9880324 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -98,6 +98,11 @@ int main(int argc, char** argv)
 
     // init
     Util::appImageEnvironment();
+    Q_INIT_RESOURCE(breeze);
+    QFile file(QLatin1String(":/light/stylesheet.qss"));
+    file.open(QFile::ReadOnly | QFile::Text);
+    QTextStream stream(&file);
+
 
 #if APPIMAGE_BUILD
     // cleanup the environment when we are running from within the AppImage
@@ -232,6 +237,7 @@ int main(int argc, char** argv)
         QGuiApplication::setWindowIcon(QIcon(QStringLiteral(":/images/icons/128-apps-hotspot.png")));
         setupDockWidgets();
         window = new MainWindow();
+        guiApp->setStyleSheet(stream.readAll());
     }
 
     const auto originalArguments = QCoreApplication::arguments();
